pkg/kubectl/describe.go

```diff
@@ -707,13 +709,14 @@ func (d *PersistentVolumeClaimDescriber) Describe(namespace, name string) (strin
 	})
 }
 
-func describeContainers(pod *api.Pod, out io.Writer) {
+// DescribeContainers is exported for consumers in other API groups that have container templates
+func DescribeContainers(containers []api.Container, containerStatuses []api.ContainerStatus, resolverFn EnvVarResolverFunc, out io.Writer) {
```

Comment:
The problem with this is that it's not really reusable across higher-level resources like replication controllers, deployments, etc.. When describing a deployment, the status of its containers is reflected in the deployment status (albeit higher level info). We still need the spec part from here though since it's part of the pod template. Ideally, this helper wouldn't print any status details if `containerStatuses` was nil.